stl/inc/utility

```diff
@@ -644,6 +644,88 @@ template <size_t _Idx>
 inline constexpr in_place_index_t<_Idx> in_place_index{};
 #endif // _HAS_CXX17
 
+#if _HAS_CXX20
+template <class _Ty>
+inline constexpr bool _Is_integral_not_bool_char = is_integral_v<_Ty>
+    && !_Is_any_of_v<_Ty, bool, char,
+#ifdef _NATIVE_WCHAR_T_DEFINED
+        wchar_t,
+#endif // _NATIVE_WCHAR_T_DEFINED
+#ifdef __cpp_char8_t
+        char8_t,
+#endif // __cpp_char8_t
+        char16_t, char32_t>;
+
+// FUNCTION TEMPLATE cmp_equal
+template <class _Ty1, class _Ty2>
+constexpr bool cmp_equal(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    static_assert(_Is_integral_not_bool_char<_Ty1> && _Is_integral_not_bool_char<_Ty2>,
+        "The integer comparison functions only accept standard and extended integer types.");
+    if constexpr (is_signed_v<_Ty1> == is_signed_v<_Ty2>) {
+        return _Left == _Right;
+    } else if constexpr (is_signed_v<_Ty2>) {
+        return _Left == static_cast<make_unsigned_t<_Ty2>>(_Right) && _Right >= 0;
+    } else {
+        return static_cast<make_unsigned_t<_Ty1>>(_Left) == _Right && _Left >= 0;
+    }
+}
+
+// FUNCTION TEMPLATE cmp_not_equal
+template <class _Ty1, class _Ty2>
+constexpr bool cmp_not_equal(const _Ty1 _Left, const _Ty2 _Right) noexcept {
+    return !cmp_equal(_Left, _Right);

```

Comment:
This non-`_Ugly` function call should be `_STD` qualified, and this issue occurs below. (You ultimately have a `static_assert` restricting this to standard integer types, but (1) ADL hijacking might happen before that is reached, and (2) our convention is to `_STD` qualify all non-`_Ugly` function calls even when they're theoretically safe.)

Reply:
Fix in a new commit, could you please check?

Reply:
Yep, looks good!